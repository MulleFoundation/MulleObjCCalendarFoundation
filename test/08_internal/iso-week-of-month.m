#ifdef __MULLE_OBJC__
# import <MulleObjCCalendarFoundation/MulleObjCCalendarFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif


int   main( void)
{
   NSCalendar                         *calendar;
   NSDateComponents                   *components;
   NSDate                             *date;
   NSInteger                          day;
   NSInteger                          month;
   NSInteger                          year;
   struct MulleExtendedTimeInterval   extended;

   calendar = [NSCalendar calendarWithIdentifier:NSGregorianCalendar];
   [calendar setTimeZone:[NSTimeZone timeZoneWithAbbreviation:@"GMT"]];
   [calendar setFirstWeekday:2];
   [calendar setMinimumDaysInFirstWeek:4];

   for( year = 2009; year <= 2020; year++)
     for( day = 30; day <= 24+7+7+7; day++)
      {
         components = [[NSDateComponents new] autorelease];
         [components setYear:year];
         [components setMonth:12];
         [components setDay:day];

         date       = [calendar dateFromComponents:components];
         components = [calendar components:~0
                                  fromDate:date];
         MulleExtendedTimeIntervalInit( &extended, [date timeIntervalSinceReferenceDate]);

         printf( "%ld.%ld.%ld ",
               (long) [components day], (long) [components month], (long) [components year]);

         printf( " week: %ld\n",
                  [calendar mulleISOWeekOfYearFromExtendedTimeInterval:&extended]);
      }
   return( 0);
}