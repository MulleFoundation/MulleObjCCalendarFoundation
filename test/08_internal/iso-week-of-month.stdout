30.12.2009  week: 53
31.12.2009  week: 53
1.1.2010  week: 53
2.1.2010  week: 53
3.1.2010  week: 53
4.1.2010  week: 1
5.1.2010  week: 1
6.1.2010  week: 1
7.1.2010  week: 1
8.1.2010  week: 1
9.1.2010  week: 1
10.1.2010  week: 1
11.1.2010  week: 2
12.1.2010  week: 2
13.1.2010  week: 2
14.1.2010  week: 2
30.12.2010  week: 52
31.12.2010  week: 52
1.1.2011  week: 52
2.1.2011  week: 52
3.1.2011  week: 1
4.1.2011  week: 1
5.1.2011  week: 1
6.1.2011  week: 1
7.1.2011  week: 1
8.1.2011  week: 1
9.1.2011  week: 1
10.1.2011  week: 2
11.1.2011  week: 2
12.1.2011  week: 2
13.1.2011  week: 2
14.1.2011  week: 2
30.12.2011  week: 52
31.12.2011  week: 52
1.1.2012  week: 52
2.1.2012  week: 1
3.1.2012  week: 1
4.1.2012  week: 1
5.1.2012  week: 1
6.1.2012  week: 1
7.1.2012  week: 1
8.1.2012  week: 1
9.1.2012  week: 2
10.1.2012  week: 2
11.1.2012  week: 2
12.1.2012  week: 2
13.1.2012  week: 2
14.1.2012  week: 2
30.12.2012  week: 52
31.12.2012  week: 1
1.1.2013  week: 1
2.1.2013  week: 1
3.1.2013  week: 1
4.1.2013  week: 1
5.1.2013  week: 1
6.1.2013  week: 1
7.1.2013  week: 2
8.1.2013  week: 2
9.1.2013  week: 2
10.1.2013  week: 2
11.1.2013  week: 2
12.1.2013  week: 2
13.1.2013  week: 2
14.1.2013  week: 3
30.12.2013  week: 1
31.12.2013  week: 1
1.1.2014  week: 1
2.1.2014  week: 1
3.1.2014  week: 1
4.1.2014  week: 1
5.1.2014  week: 1
6.1.2014  week: 2
7.1.2014  week: 2
8.1.2014  week: 2
9.1.2014  week: 2
10.1.2014  week: 2
11.1.2014  week: 2
12.1.2014  week: 2
13.1.2014  week: 3
14.1.2014  week: 3
30.12.2014  week: 1
31.12.2014  week: 1
1.1.2015  week: 1
2.1.2015  week: 1
3.1.2015  week: 1
4.1.2015  week: 1
5.1.2015  week: 2
6.1.2015  week: 2
7.1.2015  week: 2
8.1.2015  week: 2
9.1.2015  week: 2
10.1.2015  week: 2
11.1.2015  week: 2
12.1.2015  week: 3
13.1.2015  week: 3
14.1.2015  week: 3
30.12.2015  week: 53
31.12.2015  week: 53
1.1.2016  week: 53
2.1.2016  week: 53
3.1.2016  week: 53
4.1.2016  week: 1
5.1.2016  week: 1
6.1.2016  week: 1
7.1.2016  week: 1
8.1.2016  week: 1
9.1.2016  week: 1
10.1.2016  week: 1
11.1.2016  week: 2
12.1.2016  week: 2
13.1.2016  week: 2
14.1.2016  week: 2
30.12.2016  week: 52
31.12.2016  week: 52
1.1.2017  week: 52
2.1.2017  week: 1
3.1.2017  week: 1
4.1.2017  week: 1
5.1.2017  week: 1
6.1.2017  week: 1
7.1.2017  week: 1
8.1.2017  week: 1
9.1.2017  week: 2
10.1.2017  week: 2
11.1.2017  week: 2
12.1.2017  week: 2
13.1.2017  week: 2
14.1.2017  week: 2
30.12.2017  week: 52
31.12.2017  week: 52
1.1.2018  week: 1
2.1.2018  week: 1
3.1.2018  week: 1
4.1.2018  week: 1
5.1.2018  week: 1
6.1.2018  week: 1
7.1.2018  week: 1
8.1.2018  week: 2
9.1.2018  week: 2
10.1.2018  week: 2
11.1.2018  week: 2
12.1.2018  week: 2
13.1.2018  week: 2
14.1.2018  week: 2
30.12.2018  week: 52
31.12.2018  week: 1
1.1.2019  week: 1
2.1.2019  week: 1
3.1.2019  week: 1
4.1.2019  week: 1
5.1.2019  week: 1
6.1.2019  week: 1
7.1.2019  week: 2
8.1.2019  week: 2
9.1.2019  week: 2
10.1.2019  week: 2
11.1.2019  week: 2
12.1.2019  week: 2
13.1.2019  week: 2
14.1.2019  week: 3
30.12.2019  week: 1
31.12.2019  week: 1
1.1.2020  week: 1
2.1.2020  week: 1
3.1.2020  week: 1
4.1.2020  week: 1
5.1.2020  week: 1
6.1.2020  week: 2
7.1.2020  week: 2
8.1.2020  week: 2
9.1.2020  week: 2
10.1.2020  week: 2
11.1.2020  week: 2
12.1.2020  week: 2
13.1.2020  week: 3
14.1.2020  week: 3
30.12.2020  week: 53
31.12.2020  week: 53
1.1.2021  week: 53
2.1.2021  week: 53
3.1.2021  week: 53
4.1.2021  week: 1
5.1.2021  week: 1
6.1.2021  week: 1
7.1.2021  week: 1
8.1.2021  week: 1
9.1.2021  week: 1
10.1.2021  week: 1
11.1.2021  week: 2
12.1.2021  week: 2
13.1.2021  week: 2
14.1.2021  week: 2
